src/core/counter.h

```diff
@@ -0,0 +1 @@
+struct counter { uint64_t c; };
```

Comment:
Is this for the benefit of C files?  In Lua code, if you `require('core.counter')` you have this, and can use `struct counter` in any `ffi.typeof()` or `ffi.cdef[[]]`, no matter if this struct is defined on its own file or inline inside `core/counter.lua`.